esp/esdllib/esdl_script.cpp

```diff
@@ -32,9 +32,27 @@ using namespace xpp;
 interface IEsdlTransformOperation : public IInterface
 {
     virtual bool process(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) = 0;
+    virtual IInterface *prepare(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) = 0;
+    virtual bool exec(CriticalSection *crit, IInterface *stateIn, IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) = 0;
```

Comment:
Is it possible to have something more descriptive/prescriptive for the second parameter, or is it really no restrictions?